crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -2509,45 +2488,73 @@ pub(super) enum OperatorPrecedence {
 }
 
 impl OperatorPrecedence {
-    /// Returns the precedence for the given operator token or [None] if the token isn't an
-    /// operator token.
-    fn try_from_tokens(token: TokenKind, next: TokenKind) -> Option<OperatorPrecedence> {
-        Some(match token {
-            TokenKind::Or => OperatorPrecedence::Or,
-            TokenKind::And => OperatorPrecedence::And,
-            TokenKind::Not if next == TokenKind::In => {
-                OperatorPrecedence::ComparisonsMembershipIdentity
-            }
-            TokenKind::Is
-            | TokenKind::In
-            | TokenKind::EqEqual
-            | TokenKind::NotEqual
-            | TokenKind::Less
-            | TokenKind::LessEqual
-            | TokenKind::Greater
-            | TokenKind::GreaterEqual => OperatorPrecedence::ComparisonsMembershipIdentity,
-            TokenKind::Vbar => OperatorPrecedence::BitOr,
-            TokenKind::CircumFlex => OperatorPrecedence::BitXor,
-            TokenKind::Amper => OperatorPrecedence::BitAnd,
-            TokenKind::LeftShift | TokenKind::RightShift => OperatorPrecedence::LeftRightShift,
-            TokenKind::Plus | TokenKind::Minus => OperatorPrecedence::AddSub,
-            TokenKind::Star
-            | TokenKind::Slash
-            | TokenKind::DoubleSlash
-            | TokenKind::Percent
-            | TokenKind::At => OperatorPrecedence::MulDivRemain,
-            TokenKind::DoubleStar => OperatorPrecedence::Exponent,
-            _ => return None,
-        })
-    }
-
     /// Returns `true` if the precedence is right-associative i.e., the operations are evaluated
     /// from right to left.
     fn is_right_associative(self) -> bool {
         matches!(self, OperatorPrecedence::Exponent)
     }
 }
 
+#[derive(Debug)]
+enum OperatorType {
```

Comment:
Nit: We use the term `BinaryLike` for types that cover Boolean, Comparison and Binary operations. I would then name this `BinaryLikeOperator` or `AnyBinaryLikeOperator`